### DISCUSSION TITLE
Using pictures from Drive

### DISCUSSION THREAD

**Discussion by Sokhavuth** [deta/cloud-docs#439] [General]
How to embed pictures from Drive in HTML page?

**Comment by GitHub30** [comment]
You can embed pictures using `img` tag, such as in HTML


```html
<script>
  async function loadImg(e) {
    const key = "xxxxxxxxxx";
    const headers = {
      "X-Api-Key": key
    };
    const blob = await fetch(e.target.src, {
      headers
    }).then((r) => r.blob());
    e.target.src = URL.createObjectURL(blob);
  }
</script>
<img src="https://drive.deta.sh/v1/c0m8f10j/Drive/files/download?name=eevee.jpg" onerror="loadImg(event)">
```

or via Proxy

```html
<img src="https://public.deta.dev/c0m8f10j/Drive/eevee.jpg">
```

https://github.com/GitHub30/deta-drive-proxy

**Reply by Sokhavuth** [reply]
I found a solution now on how to display image from Deta Drives on html page. The solution is we can pull image from Drives as blob using Node.js or Python to include this blob into html page built on server-side before sending to client-side. Doing so, we can avoid the risk of exposing our confidential API key.

`<img src="data: image/png; base64, ${Base64.encode(blob)}" />`

**Reply by GitHub30** [reply]
Awesome!

**Reply by abdelhai** [reply]
this is not really a solution since the project key is exposed. it allows anyone to steal or change all the data in that project. please don't do this.

**Reply by awoniyosh** [reply]
hello. This is the answer.
https://github.com/orgs/deta/discussions/470
https://docs.deta.sh/docs/common_issues/

my configuration

```bash
$ tree -a -I node_modules
.
├── .DS_Store
├── .deta
│   ├── prog_info
│   └── state
├── .env
├── index.js
├── package-lock.json
├── package.json
├── public
│   ├── .DS_Store
│   ├── index.html
│   ├── script.js
│   ├── style.bk.css
│   └── style.css
└── storage
    └── logo.png

3 directories, 13 files
```

```html
// ./public/index.html
<img class="header-logo" src="../storage/logo.png"/>
```

```env
// .env
BINARY_CONTENT_TYPES=image/*,font/*
```

**Reply by Sokhavuth** [reply]
We are talking about using picture from "Drive" not from a static folder. Deta tells us only about how to upload/download pictures to/from Drive that is similar to Google Drive. But, Deta did not tell us how to use pictures from Drive in an HTML page. Moreover, Deta does not allow us to upload pictures to any static folder. It means that we can only read files but not write to files. However, we need to keep in mind that Drive is still in beta version.
[https://docs.deta.sh/docs/drive/node_tutorial](https://docs.deta.sh/docs/drive/node_tutorial)
